pyomo/contrib/parmest/tests/test_new_parmest_capabilities.py

```diff
@@ -0,0 +1,282 @@
+@unittest.skipIf(
+    not parmest.parmest_available,
+    "Cannot test parmest: required dependencies are missing",
+)
+@unittest.skipIf(not ipopt_available, "The 'ipopt' command is not available")
+class TestModelVariants(unittest.TestCase):
+
+    def setUp(self):
+        self.data = pd.DataFrame(
+            data=[[1, 8.3], [2, 10.3], [3, 19.0], [4, 16.0], [5, 15.6], [7, 19.8]],
+            columns=["hour", "y"],
+        )
+
+        # create the Rooney-Biegler model
+        def rooney_biegler_model():
+            """
+            Formulates the Pyomo model of the Rooney-Biegler example
+
+            Returns:
+                m: Pyomo model
+            """
+            m = pyo.ConcreteModel()
+
+            m.asymptote = pyo.Var(within=pyo.NonNegativeReals, initialize=10)
+            m.rate_constant = pyo.Var(within=pyo.NonNegativeReals, initialize=0.2)
+
+            m.hour = pyo.Var(within=pyo.PositiveReals, initialize=0.1)
+            m.y = pyo.Var(within=pyo.NonNegativeReals)
+
+            @m.Constraint()
+            def response_rule(m):
+                return m.y == m.asymptote * (1 - pyo.exp(-m.rate_constant * m.hour))
+
+            return m
+
+        # create the Experiment class
+        class RooneyBieglerExperiment(Experiment):
+            def __init__(self, experiment_number, hour, y):
+                self.y = y
+                self.hour = hour
+                self.experiment_number = experiment_number
+                self.model = None
+
+            def get_labeled_model(self):
+                if self.model is None:
+                    self.create_model()
+                    self.finalize_model()
+                    self.label_model()
+                return self.model
+
+            def create_model(self):
+                m = self.model = rooney_biegler_model()
+
+                return m
+
+            def finalize_model(self):
+                m = self.model
+
+                # fix the input variable
+                m.hour.fix(self.hour)
+
+                return m
+
+            def label_model(self):
+                m = self.model
+
+                # add experiment inputs
+                m.experiment_inputs = pyo.Suffix(direction=pyo.Suffix.LOCAL)
+                m.experiment_inputs.update([(m.hour, self.hour)])
+
+                # add experiment outputs
+                m.experiment_outputs = pyo.Suffix(direction=pyo.Suffix.LOCAL)
+                m.experiment_outputs.update([(m.y, self.y)])
+
+                # add unknown parameters
+                m.unknown_parameters = pyo.Suffix(direction=pyo.Suffix.LOCAL)
+                m.unknown_parameters.update(
+                    (k, pyo.value(k)) for k in [m.asymptote, m.rate_constant]
+                )
+
+                # add measurement error
+                m.measurement_error = pyo.Suffix(direction=pyo.Suffix.LOCAL)
+                m.measurement_error.update([(m.y, None)])
+
+                return m
```

Comment:
I don't think all of this should live in `setUp`. These will be executed before every single test method, so it's adding overhead that's unnecessary. It'd be better to take them out of the `setUp` method or put them above the `TestModelVariable` class. Something that doesn't make these be re-created with every test.